packages/react-ui/src/app/builder/builder-hooks.ts

```diff
@@ -135,14 +142,28 @@ export type BuilderState = {
   setSelectedNodes: (nodes: string[]) => void;
   panningMode: 'grab' | 'pan';
   setPanningMode: (mode: 'grab' | 'pan') => void;
-  pieceSelectorStep: string | null;
-  setPieceSelectorStep: (step: string | null) => void;
   isFocusInsideListMapperModeInput: boolean;
   setIsFocusInsideListMapperModeInput: (
     isFocusInsideListMapperModeInput: boolean,
   ) => void;
   isPublishing: boolean;
   setIsPublishing: (isPublishing: boolean) => void;
+  handleAddingOrUpdatingStep: (props: {
+    pieceSelectorItem: PieceSelectorItem;
+    operation: PieceSelectorOperation;
+    settings?: Record<string, unknown>;
+    selectStepAfter: boolean;
+    customLogoUrl?: string;
+  }) => string;
+
+  //Piece selector state
+  openedPieceSelectorId: string | null;
+  /**the id is either the step name or the add button id */
+  setOpenedPieceSelectorId: (id: string | null) => void;
```

Comment:
I have spent quite sometime trying to understand what is id, then I came accross this comment after 5 minutes, wouldn't it better just to name it

stepNameOrButtonId? and remove comments.

Comment should ideally never exists in the code base, sometime must for very weird cases but in generla you should just name things in declerative way.